### ISSUE TITLE
PHP 8.1 deprecations: passing null to non-nullable parameters of built-in functions

### ISSUE THREAD

**Issue by Elysiome** [google/recaptcha#513]
**Issue description**

Since PHP 8.1, the code of recaptcha emits some deprecations because passing null to non-nullable parameters of built-in functions is not allowed anymore. You can read [the article in the PHP documentation that describes these deprecations](https://www.php.net/manual/en/migration81.deprecated.php#migration81.deprecated.core.null-not-nullable-internal) if you want more info. 

Below are the logs:
```log
[21-Jan-2023 12:03:48 UTC] PHP Deprecated: strcasecmp(): Passing null to parameter #2 ($string2) of type string is deprecated in C:\project\vendor\google\recaptcha\src\ReCaptcha\ReCaptcha.php on line 168
[21-Jan-2023 12:03:48 UTC] PHP Deprecated: strcasecmp(): Passing null to parameter #2 ($string2) of type string is deprecated in C:\project\vendor\google\recaptcha\src\ReCaptcha\ReCaptcha.php on line 176
[21-Jan-2023 12:03:48 UTC] PHP Deprecated: strtotime(): Passing null to parameter #1 ($datetime) of type string is deprecated in C:\project\vendor\google\recaptcha\src\ReCaptcha\ReCaptcha.php on line 185
```

Below are the related files:
- https://github.com/google/recaptcha/blob/ed5645e799e43afa9eb181f214dc52f22982682d/src/ReCaptcha/ReCaptcha.php#L168
- https://github.com/google/recaptcha/blob/ed5645e799e43afa9eb181f214dc52f22982682d/src/ReCaptcha/ReCaptcha.php#L176
- https://github.com/google/recaptcha/blob/ed5645e799e43afa9eb181f214dc52f22982682d/src/ReCaptcha/ReCaptcha.php#L185

**Environment**

 * OS name and version: `Windows_NT x64 v10.0.19044`
 * PHP version: `v8.1`
 * Web server name and version:  `Apache v2.4.52 (Win64)`
 * `google/recaptcha` version: `v1.2`
 * Browser name and version: `Google Chrome v109.0.5414.75`

**Reproducing the issue**

Simply use recaptcha. I couldn't reproduce the issue, since it went into production and I haven't been able to reproduce it since.

**Possible solution**

A quick solution would be to cast the arguments into strings before passing them to the functions:

```diff
-if (isset($this->hostname) && strcasecmp($this->hostname, $initialResponse->getHostname()) !== 0) {
+if (isset($this->hostname) && strcasecmp($this->hostname, (string) $initialResponse->getHostname()) !== 0) {
```

```diff
-if (isset($this->action) && strcasecmp($this->action, $initialResponse->getAction()) !== 0) { 
+if (isset($this->action) && strcasecmp($this->action, (string) $initialResponse->getAction()) !== 0) { 
```

```diff
-$challengeTs = strtotime($initialResponse->getChallengeTs());
+$challengeTs = strtotime((string) $initialResponse->getChallengeTs());
```

**Comment by 8ctopus** [comment]
should be fixed with PR #490

**Comment by rowan-m** [comment]
Merged and released.
